api/include/opentelemetry/trace/propagation/HttpTraceContext.cc

```diff
@@ -0,0 +1,183 @@
+#include <vector>
```

Comment:
vector, regex are not to be used in API. Also consider the flow for when the SDK is built with no exceptions support.

Reply:
Should I use nostd::span to replace vector here?